### DISCUSSION TITLE
DiceCELoss gets 0 CE component for binary segmentation

### DISCUSSION THREAD

**Discussion by kretes** [Project-MONAI/MONAI#6919] [General]
I'm using MONAI for binary 3D segmentation.

I've noticed that in my case the 'CE' component in DiceCELoss is 0 and therefore has no effect, however I wouldn't know that if I didn't dig deep and debug the code.

repro script:
```
import torch
from monai.losses.dice import DiceLoss, DiceCELoss

shape = (2,1,3)

label = torch.randint(2, shape).type(torch.float)
pred = torch.rand(shape, requires_grad=True)

dceloss = DiceCELoss(include_background=True, sigmoid=True, lambda_ce=1)

dice_loss = dceloss.dice(pred, label)
ce_loss = dceloss.ce(pred, label)

loss = dceloss.lambda_dice * dice_loss + dceloss.lambda_ce * ce_loss

print("total", dceloss(pred, label), loss)
print("dice", dice_loss)
print("ce", ce_loss)
```
This is basically extracted from `forward` of DCELoss here https://github.com/Project-MONAI/MONAI/blob/be4e1f59cd8e7ca7a5ade5adf1aab16642c39306/monai/losses/dice.py#L723

I think what's going on here is CELoss is not aimed for binary case. 
However - DiceCELoss isn't shouting at me that I'm doing something wrong, and at the same time it gave me confidence I can use it for a single-channel case (e.g. because it gives some warnings about doing single-channel e.g. `single channel prediction, `include_background=False` ignored.` .).
Am I right that it should either be:
 - shout at the user that that DiceCELoss can't be used in single-channel scenario
 - handle this scenario internally using BCE?

**Comment by KumoLiu** [comment]
Fixed in https://github.com/Project-MONAI/MONAI/pull/6924.
